opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/EnhanceLinkdRestApi.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2008-2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2.api;
+
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.opennms.web.rest.model.v2.BridgeLinkNodeDTO;
+import org.opennms.web.rest.model.v2.CdpElementNodeDTO;
+import org.opennms.web.rest.model.v2.CdpLinkNodeDTO;
+import org.opennms.web.rest.model.v2.EnlinkdDTO;
+import org.opennms.web.rest.model.v2.IsisElementNodeDTO;
+import org.opennms.web.rest.model.v2.IsisLinkNodeDTO;
+import org.opennms.web.rest.model.v2.LldpElementNodeDTO;
+import org.opennms.web.rest.model.v2.LldpLinkNodeDTO;
+import org.opennms.web.rest.model.v2.OspfElementNodeDTO;
+import org.opennms.web.rest.model.v2.OspfLinkNodeDTO;
+
+@Path("enlinkd")
+public interface EnhanceLinkdRestApi {
+
+    @GET
+    @Path("{nodeId}")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    EnlinkdDTO getEnlinkd(@PathParam("nodeId") int nodeId);
```

Comment:
Need to update all PathParams to be strings as it could be lookupCriteria  ( either nodeId or fs:fid), Refer to  nodeDao.get(String lookupCriteria)

Reply:
This is actually fine. This calls the following code in AbstractDaoHibernate.java but not the one in the nodeDao.

    @Override
    public T get(final K id) throws DataAccessException {
        return m_entityClass.cast(getHibernateTemplate().get(m_entityClass, id));
    }

Reply:
Think I didn't explain properly.  There are two ways in which a node can be queried.  One is through id(Integer), other is through (foreignSource:foreignId).  [This](https://github.com/OpenNMS/opennms/blob/a5a028ec516ccb3cdeafdf4055232016148e9bfe/opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/NodeDao.java#L56) covers both cases.   We need to handle both query types not just with id(Integer).

Reply:
Thanks. Just updated the code just now. Please check.